src/test/rustdoc-ui/issue-74134.rs

```diff
@@ -26,12 +26,14 @@ pub struct Public {
 }
 
 // The following cases are identical to the ones above, except that they are in a private
-// module. Thus they all fall into cases 3 and 4 and should not produce a warning.
+// module. They fall either into the cases 1/2, when --document-private-items is passed, or 3/4
+// when it is not passed.
 
 mod private {
     pub struct Public {
         /// [`super::PublicType`]
         /// [`super::PrivateType`]
+        //[private]~^ WARNING public documentation for `public_item` links to a private item
```

Comment:
Hold on, this doesn't look right. This item only shows up when you pass `--document-private-items`, so it shouldn't count as public. I'm not sure if there's an easy way to tell if this is exported though.

Reply:
Tbh I stumbled over this a couple of times as well, but somehow convinced myself that it is correct this way :roll_eyes:. But now that you mention it, the item is obviously always private, regardless of `--document-private-items`.

I pushed a fixup, which changes the logic again, this time to not rely on the `item`'s visibility anymore. That removes this particular warning and keeps passing all other tests as well.